syntax = "proto2";

option java_package = "ca.eandb.jmist.engine.proto";
option java_outer_classname = "RenderEngineProtos";

import "render.proto";

message Position {
  required uint32 x = 1;
  required uint32 y = 2;
}

message Rectangle {
  required Position position = 1;
  required Size size = 2;
}

message RenderRequest {
  optional RenderJob job = 1;
  optional uint32 threads = 2;


}

message RenderCallback {
  optional bool done = 1;
  optional double progress = 2;
  optional string error = 3;

  message DrawTile {
    optional Rectangle tile = 2;

    enum PixelFormat {
      DOUBLE_RGBA = 1;
    }
    optional PixelFormat format = 3;

    optional bytes data = 4;
    optional uint64 offset = 5;
    optional uint64 length = 6;
  }
  optional DrawTile draw_tile = 4;
}
